layer/profiles.cpp

```diff
@@ -3688,6 +3917,153 @@ bool JsonLoader::GetFormat(const Json::Value &formats, const std::string &format
+            qfp.queueFamilyProperties.minImageTransferGranularity.height = minImagetransferGranularity["height"].asUInt();
+            qfp.queueFamilyProperties.minImageTransferGranularity.depth = minImagetransferGranularity["depth"].asUInt();
+            dest->push_back(qfp);
+        } else if (name == "VkQueueFamilyGlobalPriorityPropertiesKHR" || name == "VkQueueFamilyGlobalPriorityPropertiesEXT") {
+            VkQueueFamilyGlobalPriorityPropertiesKHR qfgpp = {};
+            qfgpp.sType = VK_STRUCTURE_TYPE_QUEUE_FAMILY_GLOBAL_PRIORITY_PROPERTIES_KHR;
+            uint32_t i = 0;
+            for (const auto &feature : props["priorities"]) {
+                qfgpp.priorities[i++] = StringToVkQueueGlobalPriority(feature.asString());
+            }
+            qfgpp.priorityCount = props["priorityCount"].asUInt();
+            *gp_dest = qfgpp;
+        } else if (name == "VkVideoQueueFamilyProperties2KHR") {
+            VkVideoQueueFamilyProperties2KHR vqfp = {};
+            vqfp.sType = VK_STRUCTURE_TYPE_VIDEO_QUEUE_FAMILY_PROPERTIES_2_KHR;
+            for (const auto &feature : props["videoCodecOperations"]) {
+                auto b = feature.asString();
+                auto a = StringToVkVideoCodecOperationFlags(feature.asString());
+                vqfp.videoCodecOperations |= StringToVkVideoCodecOperationFlags(feature.asString());
+            }
+            *vqfp_dest = vqfp;
+        } else if (name == "VkQueueFamilyCheckpointProperties2NV") {
+            VkQueueFamilyCheckpointProperties2NV qfcp = {};
+            qfcp.sType = VK_STRUCTURE_TYPE_QUEUE_FAMILY_CHECKPOINT_PROPERTIES_2_NV;
+            for (const auto &feature : props["checkpointExecutionStageMask"]) {
+                qfcp.checkpointExecutionStageMask |= StringToVkPipelineStageFlags2(feature.asString());
+            }
+            *qfcp2_dest = qfcp;
+        } else if (name == "VkQueueFamilyCheckpointPropertiesNV") {
+            VkQueueFamilyCheckpointPropertiesNV qfcp = {};
+            qfcp.sType = VK_STRUCTURE_TYPE_QUEUE_FAMILY_CHECKPOINT_PROPERTIES_2_NV;
+            for (const auto &feature : props["checkpointExecutionStageMask"]) {
+                qfcp.checkpointExecutionStageMask |= StringToVkPipelineStageFlags(feature.asString());
+            }
+            *qfcp_dest = qfcp;
+        } else if (name == "VkQueueFamilyQueryResultStatusProperties2KHR") {
+            VkQueueFamilyQueryResultStatusProperties2KHR qfqrsp = {};
+            qfqrsp.sType = VK_STRUCTURE_TYPE_QUEUE_FAMILY_QUERY_RESULT_STATUS_PROPERTIES_2_KHR;
+            qfqrsp.supported = props["supported"].asBool();
+            *qfqrsp_dest = qfqrsp;
+        }
+    }
+
+    bool valid = true;
+
+    for (const auto& profile_qfp : *dest) {
+        bool supported = false;
+        for (const auto& device_qfp : pdd_.device_queue_family_properties_) {
+            if (QueueFamilyMatch(device_qfp.queueFamilyProperties, profile_qfp.queueFamilyProperties)) {
```

Comment:
I don't see you checking anywhere that the extension properties match, you only check the base structure, while in fact the extension properties may define additional requirements, not just data that should be reported back.

Reply:
The way I see it is that the extensions are separate from the actual queue family properties, and I check the values below this code.

Reply:
No, this is the same problem as the one you've misunderstood above. Each base structure can have its own extension chain, so you actually have to check each chain in each iteration separately.